ui/src/app/views/workspace-edit/component.ts

```diff
@@ -56,9 +57,25 @@ export class WorkspaceEditComponent implements OnInit {
       } else {
         this.buttonClicked = true;
         this.valueNotEntered = false;
-        this.errorHandlingService.retryApi(this.workspacesService.createWorkspace(this.workspace))
-            .subscribe(cohorts => this.router.navigate(['../..'], {relativeTo : this.route}));
+        this.errorHandlingService.retryApi(
+          this.workspacesService.createWorkspace(this.workspace))
+            .subscribe(
+              (workspace) => {
+                this.navigateBack();
+              },
+              (error) => {
+                this.workspaceCreationError = true;
+              });
       }
     }
   }
+
+  navigateBack(): void {
+    this.router.navigate(['../..'], {relativeTo : this.route});
```

Comment:
I think the Location service would be better here; for now everyone will get to the new-workspace page from the same place, but this should future-proof it without much effort (if it works as advertized): https://stackoverflow.com/questions/35446955/how-to-go-back-last-page-in-angular-2 .

Reply:
I'll take a look at this!